### PR TITLE
Add help.text for the persist module tooling

### PR DESCRIPTION
**Pull request by daneshk**
## Purpose
$subject. This PR is to add ballerina-persist.help file for the persist module tooling and add entry in ballerina-help.help file.

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by rdulmina** [comment]
@Danesh Kuruppu build is failing can u check

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+OPTIONS
+       -h, --help
+           Print the usage details of all commands.
+
+BALLERINA COMMANDS
+       The below is a list of available subcommands:
+
+       init            Initialize the package for persistence
+       generate        Generate the client API based on the data model defined in the "persist" directory
```
Comment (sahanHe) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help:48]: If we change the description to include the SQL file we might also have to do it here

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+DESCRIPTION
+       Manage the data persistence in a Ballerina package.
+
+       Initializing the package for persistence is the first step toward achieving
+       data persistence in a Ballerina package. This step creates the required
+       directories and configuration files. For example:
+
+           $ bal persist init
+
+       Define the application's data model as the next step. Use the generated
```
Comment (sahanHe) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help:16]: `applications` ?

**Review by sahanHe — APPROVED** [review]

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help [hunk 3]
```diff
@@ -0,0 +1,53 @@
+NAME
+       bal persist - Manage data persistence
+
+SYNOPSIS
+       bal persist <command> [<-h> | <--help>]
+
+DESCRIPTION
+       Manage the data persistence in a Ballerina package.
```
Comment (praneesha) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help:8]:
```suggestion
       Manage data persistence in a Ballerina package.
```

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+       Initializing the package for persistence is the first step toward achieving
```
Comment (praneesha) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help:10]:
```suggestion
       Initializing the package for persistence is the first step towards achieving
```

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+       data persistence in a Ballerina package. This step creates the required
+       directories and configuration files.
+       For example:
+
+           $ bal persist init
+
+       Define the application's data model as the next step. Use the generated
```
Comment (praneesha) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help:17]:
```suggestion
       Defining the application's data model is the next step. Use the generated
```

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+       definition file in the "persist" directory to define the data model
```
Comment (praneesha) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help:18]:
```suggestion
       definition file in the `persist` directory to define the data model
```

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+       with the domain-specific entities. Use the standard Ballerina record type
+       definitions to represent entities and their relationships.
```
Comment (praneesha) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help:20]:
```suggestion
       definitions to represent the entities and their relationships.
```

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+       For example:
+
+       $ bal persist generate
+
+OPTIONS
+       -h, --help
+           Print the usage details of all commands.
+
+BALLERINA COMMANDS
+       The below is a list of available subcommands:
```
Comment (praneesha) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help:48]:
```suggestion
       The available subcommands are listed below.
```

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+
+       init            Initialize the package for persistence
+       generate        Generate the client API and SQL script based on the data model defined in the "persist" directory
```
Comment (praneesha) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help:51]:
```suggestion
       generate        Generate the client API and SQL script based on the data model defined in the `persist` directory
```

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+       bal persist - Manage data persistence
+
+SYNOPSIS
+       bal persist <command> [<-h> | <--help>]
+
+DESCRIPTION
+       Manage the data persistence in a Ballerina package.
+
+       Initializing the package for persistence is the first step toward achieving
+       data persistence in a Ballerina package. This step creates the required
```
Comment (praneesha) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help:11]:
```suggestion
       data persistence in a Ballerina package. This step creates the required
```
Shall we make the steps numbered bullet points?

Comment (daneshk) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-persist.help:11]: As this is a `help` description in the shell, better if we go with paragraph form

**Pull request merged**